issuer/src/main.rs

```diff
@@ -745,67 +758,135 @@ fn get_derivation_origin(_hostname: &str) -> Result<DerivationOriginData, Deriva
     })
 }
 
+fn args_to_string(spec: &CredentialSpec) -> String {
```

Comment:
I don't think this will be very user friendly. Why not hard code a message per current supported group type?